### PR TITLE
Refactors trnsys tests

### PR DESCRIPTION
**Pull request by louisleroy5**
Refactors tests to improve coverage and for tests to run faster

### CODE REVIEW

**Comment by samuelduchesne** [comment]
Addresses issue #68

File: tests/test_trnsys.py [hunk 1]
```diff
@@ -619,6 +620,56 @@ def test_load_idf_file_and_clean_names(self, config, converttesteasy):
...
+                    Variable_Name="Zone Thermostat Cooling Setpoint Temperature",
+                    Reporting_Frequency="hourly",
+                    save=True,
+                ),
+            },
+        ]
+
+        idf = add_object_and_run_ep(ep_version, idf_file, weather_file, outputs)
+
+        assert type(idf) == ar.idfclass.IDF
```
Comment (samuelduchesne) [inline tests/test_trnsys.py:671]: use `assert isinstance(idf, ar.idfclass.IDF)`

File: archetypal/trnsys.py [hunk 2]
```diff
@@ -425,6 +414,23 @@ def convert_idf_to_trnbuild(
     return return_path
 
 
+def add_object_and_run_ep(ep_version, idf_file, weather_file, outputs):
```
Comment (samuelduchesne) [inline archetypal/trnsys.py:417]: Why did you wrap the `run_eplus` method in this new `add_object_and_run_ep()`?

File: tests/test_trnsys.py [hunk 3]
```diff
@@ -395,6 +395,68 @@ def test_t_initial_to_b18(self, config, converttesteasy):
...
+                window_lib,
+                output_folder,
+                trnsidf_exe,
+                template,
+            ) = _assert_files(
+                idf_file, weather_file, window_lib, output_folder, trnsidf_exe, template
+            )
+        except:
+            output_folder = os.path.relpath(settings.data_folder)
+            print("Could not assert all paths exist - OK for this test")
```
Comment (samuelduchesne) [inline tests/test_trnsys.py:426]: What can cause this test to fail?

File: archetypal/trnsys.py (hunk 2 above)
Comment (louisleroy5) [reply archetypal/trnsys.py:417]: To have better coverage for trnsys.py

File: tests/test_trnsys.py (hunk 3 above)
Comment (louisleroy5) [reply tests/test_trnsys.py:426]: If the assertion do not pass

File: tests/test_trnsys.py (hunk 1 above)
Comment (louisleroy5) [reply tests/test_trnsys.py:671]: Done

**Comment by samuelduchesne** [comment]
Ok but if an assertion does not pass, you would like the test to fail, right?

Samuel

**Comment by louisleroy5** [comment]
Oh ok! But it fails if the assertion does not pass, doesn’t it ?

**Comment by samuelduchesne** [comment]
Yes. This is the role of assertions. “If assertion fails, test fails”

Samuel

**Comment by louisleroy5** [comment]
Yes and there is assertions for this test :

line 455 of test_trnsys.py : 
# Asserts closest coords
assert x == -5
assert y == 215
assert z == 0

File: tests/test_trnsys.py (hunk 3 above)
Comment (samuelduchesne) [reply tests/test_trnsys.py:425]: If you know a certain bit of code will fail, it is better to put it inside a `with pytest.raises(Exception):`

File: tests/test_trnsys.py [hunk 4]
```diff
@@ -119,14 +127,597 @@ def test_get_save_write_schedules_as_input(self, config, converttest):
...
+                idf_file,
+                weather_file,
+                window_lib,
+                output_folder,
+                trnsidf_exe,
+                template,
+            ) = _assert_files(
+                idf_file, weather_file, window_lib, output_folder, trnsidf_exe, template
+            )
+        except:
```
Comment (samuelduchesne) [inline tests/test_trnsys.py:317]: This except statement is too general. try, except what? KeyError? ValueError?

Comment (samuelduchesne) [reply tests/test_trnsys.py:317]: Even better, put the `_assert_files` under a `with pytest.raises(Exception):` and get rid of the try, except clause

Make sure to replace "Exception" by the actual exception your method raises. This way, pytest will be happy and display the correct information

Comment (samuelduchesne) [reply tests/test_trnsys.py:317]: See https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions for more info

**Pull request merged**